File: domain/src/test/java/no/nav/melosys/domain/util/LovvalgBestemmelseUtilsTest.java
package no.nav.melosys.domain.util;

import no.nav.melosys.domain.kodeverk.LovvalgBestemmelse;
import no.nav.melosys.domain.kodeverk.lovvalgsbestemmelser.Lovvalgbestemmelser_trygdeavtale_uk;
import org.junit.jupiter.api.Test;
import static org.assertj.core.api.Assertions.assertThat;

import static org.assertj.core.api.Assertions.assertThatExceptionOfType;

class LovvalgBestemmelseUtilsTest {

    @Test
    void dbDataTilLovvalgBestemmelseIkkeFunnet() {
        assertThatExceptionOfType(IllegalArgumentException.class)
            .isThrownBy(() -> LovvalgBestemmelseUtils.dbDataTilLovvalgBestemmelse("test"))
            .withMessage("Lovvalgbestemmelse kode:test ikke funnet");
    }

    @Test
    void dbDataTilLovvalgBestemmelse() {
        LovvalgBestemmelse lovvalgBestemmelse = LovvalgBestemmelseUtils.dbDataTilLovvalgBestemmelse("UK_ART7");
        assertThat(lovvalgBestemmelse.getKode()).isEqualTo("UK_ART7");
        assertThat(lovvalgBestemmelse).isInstanceOf(Lovvalgbestemmelser_trygdeavtale_uk.class);
    }
}

File: domain/src/test/kotlin/no/nav/melosys/domain/util/LovvalgBestemmelseUtilsTest.kt
package no.nav.melosys.domain.util

import io.kotest.assertions.throwables.shouldThrow
import io.kotest.matchers.shouldBe
import io.kotest.matchers.types.shouldBeInstanceOf
import no.nav.melosys.domain.kodeverk.lovvalgsbestemmelser.trygdeavtale.Lovvalgsbestemmelser_trygdeavtale_gb
import org.junit.jupiter.api.Test

internal class LovvalgBestemmelseUtilsTest {

    @Test
    fun dbDataTilLovvalgBestemmelseIkkeFunnet() {
        val exception = shouldThrow<IllegalArgumentException> {
            LovvalgBestemmelseUtils.dbDataTilLovvalgBestemmelse("test")
        }
        exception.message shouldBe "Lovvalgbestemmelse kode:test ikke funnet"
    }

    @Test
    fun dbDataTilLovvalgBestemmelse() {
        val lovvalgBestemmelse = LovvalgBestemmelseUtils.dbDataTilLovvalgBestemmelse("UK_ART7_3")


        lovvalgBestemmelse.run {
            kode shouldBe "UK_ART7_3"
            shouldBeInstanceOf<Lovvalgsbestemmelser_trygdeavtale_gb>()
        }
    }
}
